src/app/components/account-dialog/account-dialog.component.ts

```diff
@@ -0,0 +1,268 @@
+
+    private ngUnsubscribe = new Subject();
+
+    constructor(
+        @Inject(MAT_DIALOG_DATA) data: AccountDialogPayload,
+        private dialogRef: MatDialogRef<AccountDialogComponent>,
+        private raidenService: RaidenService,
+        private fb: FormBuilder,
+        private tokenPollingService: TokenPollingService,
+        private userDepositService: UserDepositService,
+        private notificationService: NotificationService,
+        private web3Factory: Web3Factory
+    ) {
+        this.form = this.fb.group({
+            asset: [data.asset, Validators.required],
+            amount: ['', Validators.required],
+        });
+
+        this.ethBalance$ = raidenService.balance$;
+        this.zeroEthBalance$ = raidenService.balance$.pipe(
+            map((balance) => new BigNumber(balance).isZero())
+        );
+        this.network$ = raidenService.network$;
+        this.faucetLink$ = zip(
+            raidenService.network$,
+            raidenService.raidenAddress$
+        ).pipe(
+            map(([network, raidenAddress]) =>
+                network.faucet.replace('${ADDRESS}', raidenAddress)
+            )
+        );
+    }
+
+    ngOnInit(): void {
+        const servicesToken$ = this.userDepositService.servicesToken$.pipe(
+            first()
+        );
+        combineLatest([this.form.controls.asset.valueChanges, servicesToken$])
+            .pipe(
+                filter(([asset, servicesToken]) => asset !== 'ETH'),
+                switchMap(([selectedToken, servicesToken]) => {
+                    if (selectedToken.address === servicesToken.address) {
+                        return this.userDepositService.servicesToken$;
+                    }
+                    return this.tokenPollingService.getTokenUpdates(
+                        selectedToken.address
+                    );
+                }),
+                map((token) => token.balance),
+                takeUntil(this.ngUnsubscribe)
+            )
+            .subscribe((balance) => (this.balance = balance));
+
+        this.form.controls.asset.valueChanges
+            .pipe(takeUntil(this.ngUnsubscribe))
+            .subscribe((asset: UserToken | 'ETH') => {
+                this.tokenInput.selectedToken = asset;
+                if (asset === 'ETH') {
+                    this.ethSelected = true;
+                    this.token = undefined;
+                } else {
+                    this.ethSelected = false;
+                    this.token = asset;
+                }
+            });
+        this.form.controls.asset.updateValueAndValidity();
+
+        from(this.web3Factory.detectProvider())
+            .pipe(
+                filter((provider) => !!provider),
+                map((provider: any) => this.web3Factory.create(provider)),
+                tap((web3) => (this.web3 = web3)),
+                switchMap((web3) =>
+                    from<Promise<string[]>>(web3.eth.getAccounts())
+                )
```

Comment:
```suggestion
                switchMap(async (web3) => web3.eth.getAccounts())
```
Suggestion: `*Map` can accept promises directly; if the `getAccounts()` method isn't typed properly, it's correct to use the above with `as Promise<string[]>`, if needed.
The `async` in the callback isn't required, but helps you ensure, in case the method returns a plain value instead of a Promise, it'll still be resolved as Promise instead of an unexpected behavior like unfolding an `Array<string>`

Reply:
Nice, I didn't know about this.